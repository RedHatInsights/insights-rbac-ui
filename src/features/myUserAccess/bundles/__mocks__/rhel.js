import React from 'react';

const InsightsBundle = () => <div id="rhel-mock">This is just a mock insights bundle module</div>;

export default InsightsBundle;
