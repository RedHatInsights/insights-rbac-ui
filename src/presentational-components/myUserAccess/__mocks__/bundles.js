export const bundleData = [
  {
    entitlement: 'entitled',
    title: 'Entitled',
    apps: {
      eApp1: '/eapp1',
      eApp2: '/eapp2',
    },
    appsIds: ['a1', 'a2'],
  },
  {
    entitlement: 'not-entitled',
    title: 'Not entitled',
    apps: {
      neApp1: '/neapp1',
      neApp2: '/neapp2',
    },
    appsIds: ['na1', 'na2'],
  },
];
