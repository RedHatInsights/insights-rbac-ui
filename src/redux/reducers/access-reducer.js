import { GET_PRINCIPAL_ACCESS } from '../../redux/action-types';
import { defaultSettings } from '../../helpers/shared/pagination';

// Initial State
export const accessInitialState = {
  isLoading: false,
  access: {
    data: [],
    meta: defaultSettings,
  },
};

const setLoadingState = (state) => ({ ...state, isLoading: true });

const setAccess = (state, { payload }) => ({ ...state, access: payload, isLoading: false });

export default {
  [`${GET_PRINCIPAL_ACCESS}_PENDING`]: setLoadingState,
  [`${GET_PRINCIPAL_ACCESS}_FULFILLED`]: setAccess,
};
