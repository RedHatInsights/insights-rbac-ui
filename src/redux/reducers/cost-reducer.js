import { GET_RESOURCE_DEFINITIONS, GET_RESOURCE } from '../../redux/action-types';
import { defaultSettings } from '../../helpers/shared/pagination';

// Initial State
export const costInitialState = {
  isLoading: false,
  resourceTypes: {
    data: [],
    meta: defaultSettings,
  },
  resources: {},
  loadingResources: 0,
};

const setLoadingState = (state) => ({ ...state, isLoading: true });

const setResourceDefinitions = (state, { payload }) => ({ ...state, resourceTypes: payload, isLoading: false });
const setResource = (state, { payload }) => ({
  ...state,
  resources: { ...state.resources, [payload.links.first.split('/')[5]]: payload.data },
  loadingResources: state.loadingResources - 1,
});

const setResourceLoading = (state) => ({ ...state, loadingResources: state.loadingResources + 1 });

export default {
  [`${GET_RESOURCE_DEFINITIONS}_PENDING`]: setLoadingState,
  [`${GET_RESOURCE_DEFINITIONS}_FULFILLED`]: setResourceDefinitions,
  [`${GET_RESOURCE}_PENDING`]: setResourceLoading,
  [`${GET_RESOURCE}_FULFILLED`]: setResource,
};
