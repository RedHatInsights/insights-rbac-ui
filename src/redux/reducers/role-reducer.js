import { FETCH_ROLE, FETCH_ROLES, FETCH_ROLE_FOR_USER, FETCH_ROLE_FOR_PRINCIPAL, FETCH_ROLES_FOR_WIZARD } from '../../redux/action-types';
import { defaultSettings } from '../../helpers/shared/pagination';

// Initial State
export const rolesInitialState = {
  isLoading: false,
  isRecordLoading: false,
  roles: {
    data: [],
    meta: defaultSettings,
  },
  rolesForWizard: {
    data: [],
    meta: defaultSettings,
  },
  selectedRole: {},
};

const setLoadingState = (state) => ({ ...state, isLoading: true });
const setRecordLoadingState = (state) => ({ ...state, isRecordLoading: true });
const setRole = (state, { payload }) => ({ ...state, selectedRole: payload, isRecordLoading: false });
const setRoles = (state, { payload }) => ({ ...state, roles: payload, isLoading: false });
const setRolesWithAccess = (state, { payload }) => ({
  ...state,
  rolesWithAccess: { ...state.rolesWithAccess, [payload.uuid]: payload },
  isRecordLoading: false,
});

const setRolesForWizard = (state, { payload }) => ({ ...state, rolesForWizard: payload, isWizardLoading: false });
const setWizardLoadingState = (state) => ({ ...state, isWizardLoading: true });

export default {
  [`${FETCH_ROLE}_FULFILLED`]: setRole,
  [`${FETCH_ROLE}_PENDING`]: setRecordLoadingState,
  [`${FETCH_ROLES}_FULFILLED`]: setRoles,
  [`${FETCH_ROLES}_PENDING`]: setLoadingState,
  [`${FETCH_ROLE_FOR_USER}_FULFILLED`]: setRolesWithAccess,
  [`${FETCH_ROLE_FOR_USER}_PENDING`]: setRecordLoadingState,
  [`${FETCH_ROLE_FOR_PRINCIPAL}_FULFILLED`]: setRolesWithAccess,
  [`${FETCH_ROLE_FOR_PRINCIPAL}_PENDING`]: setRecordLoadingState,
  [`${FETCH_ROLES_FOR_WIZARD}_FULFILLED`]: setRolesForWizard,
  [`${FETCH_ROLES_FOR_WIZARD}_PENDING`]: setWizardLoadingState,
};
