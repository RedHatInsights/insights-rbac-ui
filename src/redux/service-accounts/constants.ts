export const NO_DATA = 'no-data';
export const LAST_PAGE = 'last-page';
export const RESULTS = 'results';
export const ERROR = 'error';
