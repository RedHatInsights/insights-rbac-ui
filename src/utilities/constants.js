export const RBAC_API_BASE = `/api/rbac/v1`;
