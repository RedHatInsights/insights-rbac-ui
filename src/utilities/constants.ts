export const RBAC_API_BASE = `/api/rbac/v1` as const;
export const RBAC_API_BASE_2 = `/api/rbac/v2` as const;
export const COST_API_BASE = `/api/cost-management/v1` as const;
export const INVENTORY_API_BASE = `/api/inventory/v1` as const;

export const DEFAULT_MUA_BUNDLE = 'rhel' as const;

export const DEFAULT_ACCESS_GROUP_ID = 'default-access' as const;
