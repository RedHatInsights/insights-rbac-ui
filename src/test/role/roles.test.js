import React from 'react';
import { act } from 'react-dom/test-utils';
import { mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import { MemoryRouter, Route } from 'react-router-dom';
import { Provider } from 'react-redux';
import toJson from 'enzyme-to-json';
import promiseMiddleware from 'redux-promise-middleware';
import Roles from '../../smart-components/role/roles';
import { notificationsMiddleware } from '@redhat-cloud-services/frontend-components-notifications';
import { rolesInitialState } from '../../redux/reducers/role-reducer';
import { mock } from '../__mocks__/apiMock';
import { RBAC_API_BASE } from '../../utilities/constants';
import { getUserMock } from '../../../config/setupTests';

describe('<Roles />', () => {
  const middlewares = [promiseMiddleware, notificationsMiddleware()];
  let mockStore;
  let initialState;

  beforeEach(() => {
    mockStore = configureStore(middlewares);
    initialState = {
      roleReducer: {
        ...rolesInitialState,
        roles: {
          identity: {
            user: {},
          },
          data: [
            {
              name: 'Test name',
              uuid: 'test',
              description: 'test',
              system: 'test',
              accessCount: 'test',
              groups_in_count: 5,
              modified: new Date(0),
            },
          ],
        },
        isLoading: false,
      },
    };
    mock.onGet(`${RBAC_API_BASE}/roles/`).replyOnce(200);
  });

  it('should render correctly', () => {
    const store = mockStore(initialState);
    mock.onGet(`/api/rbac/v1/roles/?name=&scope=account&add_fields=groups_in_count`).reply(200, {});
    const wrapper = mount(
      <Provider store={store}>
        <MemoryRouter initialEntries={['/roles']}>
          <Route path="/roles" component={Roles} />
        </MemoryRouter>
      </Provider>
    );
    expect(toJson(wrapper.find('TableToolbarView'), { mode: 'shallow' })).toMatchSnapshot();
  });

  it('should render correctly in loading state', () => {
    const store = mockStore({
      ...initialState,
      roleReducer: {
        ...initialState.roleReducer,
        isLoading: true,
      },
    });
    const wrapper = mount(
      <Provider store={store}>
        <MemoryRouter initialEntries={['/roles']}>
          <Route path="/roles" component={Roles} />
        </MemoryRouter>
      </Provider>
    );
    expect(toJson(wrapper.find('ListLoader'), { mode: 'shallow' })).toMatchSnapshot();
  });

  it('should render correctly in org admin', () => {
    const store = mockStore({
      ...initialState,
      roleReducer: {
        ...initialState.roleReducer,
        roles: {
          ...initialState.roleReducer.roles,
          identity: {
            user: { is_org_admin: true },
          },
        },
      },
    });
    const wrapper = mount(
      <Provider store={store}>
        <MemoryRouter initialEntries={['/roles']}>
          <Route path="/roles" component={Roles} />
        </MemoryRouter>
      </Provider>
    );
    expect(toJson(wrapper.find('TableToolbarView'), { mode: 'shallow' })).toMatchSnapshot();
  });

  it('should fetch roles on sort click', async () => {
    const store = mockStore(initialState);
    mock.onGet(`/api/rbac/v1/roles/?name=&scope=account&add_fields=groups_in_count`).reply(200, {});
    mock.onGet(`/api/rbac/v1/roles/?limit=20&scope=account&order_by=name&add_fields=groups_in_count`).replyOnce(200, {});
    let wrapper;
    await act(async () => {
      wrapper = mount(
        <Provider store={store}>
          <MemoryRouter initialEntries={['/roles']}>
            <Route path="/roles" component={Roles} />
          </MemoryRouter>
        </Provider>
      );
    });
    store.clearActions();
    await act(async () => {
      wrapper.find('span.pf-c-table__sort-indicator').first().simulate('click');
    });
    const expectedPayloadAfter = [
      { type: 'FETCH_ROLES_PENDING' },
      {
        type: 'FETCH_ROLES_FULFILLED',
        payload: {
          ...getUserMock,
        },
      },
    ];
    expect(store.getActions()).toEqual(expectedPayloadAfter);
  });
});
