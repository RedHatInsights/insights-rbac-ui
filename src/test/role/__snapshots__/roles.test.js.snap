// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Roles /> should render correctly 1`] = `
<TableToolbarView
  actionResolver={[Function]}
  borders={true}
  columns={
    Array [
      Object {
        "key": "name",
        "title": "Name",
        "transforms": Array [
          [Function],
          [Function],
        ],
      },
      Object {
        "title": "Description",
      },
      Object {
        "title": "Permissions",
        "transforms": Array [
          [Function],
        ],
      },
      Object {
        "title": "Groups",
        "transforms": Array [
          [Function],
        ],
      },
      Object {
        "key": "modified",
        "title": "Last modified",
        "transforms": Array [
          [Function],
          [Function],
        ],
      },
    ]
  }
  createRows={[Function]}
  data={
    Array [
      Object {
        "accessCount": "test",
        "description": "test",
        "groups_in_count": 5,
        "modified": 1970-01-01T00:00:00.000Z,
        "name": "Test name",
        "system": "test",
        "uuid": "test",
      },
    ]
  }
  fetchData={[Function]}
  filterItems={Array []}
  filterPlaceholder="name"
  filterValue=""
  filters={Array []}
  hideFilterChips={false}
  isCollapsible={false}
  isCompact={false}
  isFilterable={false}
  isLoading={false}
  isSelectable={false}
  pagination={
    Object {
      "itemCount": 0,
      "limit": 20,
      "offset": 0,
    }
  }
  routes={[Function]}
  setCheckedItems={[Function]}
  setFilterValue={[Function]}
  titlePlural="roles"
  titleSingular="role"
  toolbarButtons={[Function]}
/>
`;

exports[`<Roles /> should render correctly in loading state 1`] = `
<ListLoader
  items={5}
/>
`;

exports[`<Roles /> should render correctly in org admin 1`] = `
<TableToolbarView
  actionResolver={[Function]}
  borders={true}
  columns={
    Array [
      Object {
        "key": "name",
        "title": "Name",
        "transforms": Array [
          [Function],
          [Function],
        ],
      },
      Object {
        "title": "Description",
      },
      Object {
        "title": "Permissions",
        "transforms": Array [
          [Function],
        ],
      },
      Object {
        "title": "Groups",
        "transforms": Array [
          [Function],
        ],
      },
      Object {
        "key": "modified",
        "title": "Last modified",
        "transforms": Array [
          [Function],
          [Function],
        ],
      },
    ]
  }
  createRows={[Function]}
  data={
    Array [
      Object {
        "accessCount": "test",
        "description": "test",
        "groups_in_count": 5,
        "modified": 1970-01-01T00:00:00.000Z,
        "name": "Test name",
        "system": "test",
        "uuid": "test",
      },
    ]
  }
  fetchData={[Function]}
  filterItems={Array []}
  filterPlaceholder="name"
  filterValue=""
  filters={Array []}
  hideFilterChips={false}
  isCollapsible={false}
  isCompact={false}
  isFilterable={false}
  isLoading={false}
  isSelectable={false}
  pagination={
    Object {
      "itemCount": 0,
      "limit": 20,
      "offset": 0,
    }
  }
  routes={[Function]}
  setCheckedItems={[Function]}
  setFilterValue={[Function]}
  titlePlural="roles"
  titleSingular="role"
  toolbarButtons={[Function]}
/>
`;
